lib/sidekiq/batch.rb

```diff
@@ -179,40 +179,38 @@ def process_failed_job(bid, jid)
 
       def process_successful_job(bid, jid)
         failed, pending, children, complete, success, total, parent_bid = Sidekiq.redis do |r|
-          r.multi do
-            r.scard("BID-#{bid}-failed")
-            r.hincrby("BID-#{bid}", "pending", -1)
-            r.hincrby("BID-#{bid}", "children", 0)
-            r.scard("BID-#{bid}-complete")
-            r.scard("BID-#{bid}-success")
-            r.hget("BID-#{bid}", "total")
-            r.hget("BID-#{bid}", "parent_bid")
-
-            r.srem("BID-#{bid}-failed", jid)
-            r.srem("BID-#{bid}-jids", jid)
-            r.expire("BID-#{bid}", BID_EXPIRE_TTL)
+          r.multi do |pipeline|
+            pipeline.scard("BID-#{bid}-failed")
+            pipeline.hincrby("BID-#{bid}", "pending", -1)
+            pipeline.hincrby("BID-#{bid}", "children", 0)
+            pipeline.scard("BID-#{bid}-complete")
+            pipeline.scard("BID-#{bid}-success")
+            pipeline.hget("BID-#{bid}", "total")
+            pipeline.hget("BID-#{bid}", "parent_bid")
+
+            pipeline.srem("BID-#{bid}-failed", jid)
+            pipeline.srem("BID-#{bid}-jids", jid)
+            pipeline.expire("BID-#{bid}", BID_EXPIRE_TTL)
           end
         end
 
-        all_success = pending.to_i.zero? && children == success
         # if complete or successfull call complete callback (the complete callback may then call successful)
-        if (pending.to_i == failed.to_i && children == complete) || all_success
+        if (pending.to_i == failed.to_i && children == complete) || (pending.to_i.zero? && children == success)
           enqueue_callbacks(:complete, bid)
-          enqueue_callbacks(:success, bid) if all_success
```

Comment:
What's the need behind getting rid of this?
This change isn't part of the depreciation warning fix, is it?